src/charm_state.py

```diff
@@ -213,9 +219,17 @@ def from_charm(
             )
             error_field_str = " ".join(f"flask_{f}" for f in error_fields)
             raise CharmConfigInvalidError(f"invalid configuration: {error_field_str}") from exc
+        database_migration_script = charm.config.get("database_migration_script")
```

Comment:
Couldn't this be checked on database_migration since you have a container instance that can be used to check the existence of the file? The script is supposed to be there, right?